frontends/web/src/routes/device/bitbox02/setup/wallet-create.tsx

```diff
@@ -0,0 +1,154 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import * as bitbox02 from '../../../../api/bitbox02';
+import { alertUser } from '../../../../components/alert/Alert';
+import { Wait } from './wait';
+import { ChecklistWalletCreate } from './checklist';
+import { SetDeviceName } from './name';
+import { SetPassword } from './password';
+
+type TCreateWalletStatus = 'intro' | 'setPassword' | 'createBackup';
+
+type TWait = {
+  title: string;
+  text?: string;
+};
+
+type Props = {
+  deviceID: string;
+  ensureSDCard: () => Promise<boolean>;
+  isSeeded: boolean;
+  onAbort: () => void;
+};
+
+export const CreateWallet = ({
+  deviceID,
+  ensureSDCard,
+  isSeeded,
+  onAbort,
+}: Props) => {
+  const { t } = useTranslation();
+  const [status, setStatus] = useState<TCreateWalletStatus>('intro');
+  const [errorText, setErrorText] = useState('');
+  const [waitView, setWaitView] = useState<TWait>();
+  const [hasSDCard, setSDCard] = useState<boolean>();
+
+  useEffect(() => {
+    bitbox02.checkSDCard(deviceID).then(setSDCard);
+  }, [deviceID]);
+
+  const ensurePassword = async () => {
+    setStatus('setPassword');
+    try {
+      const result = await bitbox02.setPassword(deviceID);
+      if (!result.success) {
+        if (result.code === bitbox02.errUserAbort) {
+          // On user abort, just go back to the first screen. This is a bit lazy, as we should show
+          // a screen to ask the user to go back or try again.
+          setErrorText('');
+          onAbort();
+        } else {
+          setErrorText(t('bitbox02Wizard.noPasswordMatch'));
+          ensurePassword();
+        }
+        // show error and do NOT continue to createBackup
+        return;
+      }
+      setErrorText('');
+      setStatus('createBackup');
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const setDeviceName = async (deviceName: string) => {
+    setWaitView({ title: t('bitbox02Interact.confirmName') });
+    try {
+      const result = await bitbox02.setDeviceName(deviceID, deviceName);
+      if (!result.success) {
+        alertUser(result.message || t('genericError'), {
+          asDialog: false,
+          callback: () => setWaitView(undefined),
+        });
+        return;
+      }
+      setWaitView(undefined);
+      ensurePassword();
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const createBackup = async () => {
+    try {
+      const result1 = await ensureSDCard();
+      if (!result1) {
+        alertUser(t('bitbox02Wizard.createBackupFailed'), { asDialog: false });
+        return;
+      }
+      setWaitView({
```

Comment:
For some reason, this wait view is not displayed if the sd card is unplugged while calling `ensureSDCard`.. the checklist is displayed again while the device is asking for date confirmation.

Reply:
I must have forgotten when looking at a different edgecase. thanks 

fixed in https://github.com/digitalbitbox/bitbox-wallet-app/pull/2120/commits/512f337268e873dd56c46a0c9493d26fda8f4853

will squash once all good